x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/Wildcard.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.expression.function.scalar.string;
+
+import org.elasticsearch.xpack.eql.utils.StringUtils;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.Expressions;
+import org.elasticsearch.xpack.ql.expression.Expressions.ParamOrdinal;
+import org.elasticsearch.xpack.ql.expression.function.scalar.ScalarFunction;
+import org.elasticsearch.xpack.ql.expression.gen.pipeline.Pipe;
+import org.elasticsearch.xpack.ql.expression.gen.script.ScriptTemplate;
+import org.elasticsearch.xpack.ql.expression.predicate.logical.Or;
+import org.elasticsearch.xpack.ql.expression.predicate.regex.Like;
+import org.elasticsearch.xpack.ql.tree.NodeInfo;
+import org.elasticsearch.xpack.ql.tree.Source;
+import org.elasticsearch.xpack.ql.type.DataType;
+import org.elasticsearch.xpack.ql.type.DataTypes;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.elasticsearch.common.logging.LoggerMessageFormat.format;
+import static org.elasticsearch.xpack.ql.expression.TypeResolutions.isStringAndExact;
+
+/**
+ * EQL wildcard function. Matches the form:
+ *     wildcard(field, "*wildcard*pattern*", "*wildcard*pattern*")
+ */
+
+public class Wildcard extends ScalarFunction {
+
+    private final Expression field;
+    private final List<Expression> patterns;
+
+    private static List<Expression> getArguments(Expression src, List<Expression> patterns) {
+        List<Expression> arguments = new java.util.ArrayList<>(Collections.singletonList(src));
+        arguments.addAll(patterns);
+        return arguments;
+    }
+
+    public Wildcard(Source source, Expression field, List<Expression> patterns) {
+        super(source, getArguments(field, patterns));
+        this.field = field;
+        this.patterns = patterns;
+    }
+
+    public Expression asLikes() {
```

Comment:
Since it's always a Like or an Or, return a `ScalarFunction` instead. Also the method should be private.

Reply:
Currently accessing via the Optimizer, which is why it was left public:
https://github.com/elastic/elasticsearch/pull/54020/files#diff-cb6f05d0222f3291ea3df5d33525d6f6R71